src/lib/voiceSearchHelper/index.ts

```diff
@@ -1,21 +1,32 @@
-const STATUS_INITIAL = 'initial';
-const STATUS_ASKING_PERMISSION = 'askingPermission';
-const STATUS_WAITING = 'waiting';
-const STATUS_RECOGNIZING = 'recognizing';
-const STATUS_FINISHED = 'finished';
-const STATUS_ERROR = 'error';
-
 export type VoiceSearchHelperParams = {
   searchAsYouSpeak: boolean;
   onQueryChange: (query: string) => void;
   onStateChange: () => void;
 };
 
+export type Status =
+  | 'initial'
+  | 'askingPermission'
+  | 'waiting'
+  | 'recognizing'
+  | 'finished'
+  | 'error';
+
+export type ErrorCode =
+  | 'no-speech'
+  | 'aborted'
+  | 'audio-capture'
+  | 'network'
+  | 'not-allowed'
+  | 'service-not-allowed'
+  | 'bad-grammar'
+  | 'language-not-supported';
+
 export type VoiceListeningState = {
-  status: string;
+  status: Status;
   transcript: string;
   isSpeechFinal: boolean;
-  errorCode?: string;
+  errorCode?: ErrorCode;
```

Comment:
Could be typed as a `SpeechRecognitionErrorCode`

Reply:
Right, it works fine :)